controller/node_controller.go

```diff
@@ -181,6 +212,27 @@ func (nc *NodeController) filterReplica(r *longhorn.Replica) bool {
 	return false
 }
 
+func (nc *NodeController) filterManagerPod(obj *v1.Pod) bool {
+	// only filter pod that control by manager DaemonSet to deal with physical node down
+	metaObj, err := meta.Accessor(obj)
+	if err != nil {
+		logrus.Warnf("BUG: %v cannot be convert to metav1.Object: %v", obj, err)
+		return false
+	}
+
+	ownerRefs := metaObj.GetOwnerReferences()
+	controlByManager := false
+	for _, ref := range ownerRefs {
+		if ref.Kind == daemonKind {
```

Comment:
You don't need to always assume the manager was deployed using daemon set.

Instead, you can check if the pod has a container named `longhorn-manager`.